pkg/destroy/baremetal/baremetal.go

```diff
@@ -24,17 +26,27 @@ type ClusterUninstaller struct {
 func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 	o.Logger.Debug("Deleting bare metal resources")
 
-	// FIXME: close the connection
-	conn, err := libvirt.NewConnect(o.LibvirtURI)
+	uri, err := url.Parse(o.LibvirtURI)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to connect to Libvirt daemon")
+		return nil, err
 	}
-	err = o.deleteStoragePool(conn)
+
+	virt, err := libvirt.ConnectToURI(uri)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to clean baremetal bootstrap storage pool")
+		return nil, err
 	}
+	defer func() {
+		if err := virt.Disconnect(); err != nil {
+			if o.Logger != nil {
+				o.Logger.Warn("failed to disconnect from libvirt", err)
+			}
+		}
+	}()
 
-	o.Logger.Debug("FIXME: delete resources!")
+	err = o.deleteStoragePool(virt)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to clean baremetal bootstrap storage pool")
```

Comment:
There's an ongoing effort to get rid of `errors.Wrap`, in favor to `fmt.Errorf`

Reply:
Good point. I think I was matching the style of the other file here. I'll submit a follow-up to change all of them at once.

Reply:
Had to rebase, so I added that commit in here.